specification/callbacks.md

```diff
@@ -0,0 +1,91 @@
+
+<figure style="text-align:center">
+  <object type="image/svg+xml" data="{{site.baseurl}}/img/callback-object.svg"></object>
+  <figcaption>Callbacks are implemented in an <a href="paths.html#the-operation-object">Operation</a> and also be described by reference using a Reference Object.</figcaption>
+</figure>
+
+A common use case for callbacks is to provide information on the success or failure of long-running operations that cannot be completed synchronously. An example is in the context of payments API, where payments backend systems are asynchronous and therefore cannot always return a definitive response within the typical timeout period for an HTTP request. Callbacks are therefore implemented to allow an immediate, synchronous response to be made so that the API consumer can check on the status of the payment out-of-band. This supports polling patterns that are often implemented by API providers.
+
+A Callback object is defined as part of an Operation. Our Tic Tac Toe example implements a Response object that indicates an asynchronous response to a requested move on the board. A 202 HTTP response is returned - indicating the request has been accepted but the operation has yet to complete - along with the `Location` header that provides the callback URL:
+
+```yaml
+put:
+  summary: Set a single board square
+  description: Places a mark on the board and retrieves the whole board and the winner (if any).
+  tags:
+    - Gameplay
+  operationId: put-square
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          $ref: "#/components/schemas/mark"
+  responses:
+    "200":
+      description: "OK"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/status"
+    "202":
+      description: Mark operation has not completed. Use callback to check for progress
+      headers:
+        Location:
+          description: Callback URL
+          schema:
+          type: string
+```
+
+This operation also describes the callback, which is implemented as a [Runtime Expression](https://spec.openapis.org/oas/v3.1.0#runtime-expressions) that points at the `Location` header from the 202 response. The `callbacks` object is defined as a Map. Each callback is defined with a name that uniquely identifies the Callback object, including the expression value:
+
+```yaml
+callbacks:
+  statusCallback:
```

Comment:
your documentation is great as it is confusing me ... 

i would not expect that i got a callback like this 

to me i understood the callback pattern mode like this 

if  C is the client  and S the server 

C call S on POST /long-running-Task 
but in your scenario with location it look that the task return a location 
it looks like C need to call for the status of the long running task 
in fact this would be just a dynamic resource creation that for


vs i thought that it was more that C call S POST /long-running-task 

and that the S call then back the client with this is Finished , or that the sever call in progress , then server call again Finished

is the statement "API consumer should call a callback URL" correct ? i was expecting more the serveur to call the callback URL that the Client must implement

Reply:
Yes you are right. I'll refactor

Reply:
@LasneF I've now refactored the page with the correct facts. Basically I've never used Callbacks in anger and essentially didn't read the specification correctly. Thanks for calling this out.